CommuniTEAM/CommuniTEA
# [TEA-22] Initialize Database Dev Environment (Migrations: Locations Tables)

### In this PR, I'm adding/changing the following:

- Added a migration for location tables: states and cities
- Added a migration to populate the states table with its data
- Added `.env` support to the docker containers so that only one `.env` is needed
- Changed the `.env.sample` to reflect this change so be sure to update your local `.env` to match
- Added [docker-compose-wait](https://github.com/ufoscout/docker-compose-wait) to prevent the `goapi` container from shutting down while waiting for the database to come up
- Added [Dbmate](https://github.com/amacneil/dbmate) to the `goapi` container and its migration command to the `docker-compose.yaml` so that migrations run automatically on container startup
- Made a README for the backend that details setup and development instructions for the database
- Added a [SQLFluff](https://docs.sqlfluff.com/en/stable/index.html) as a pre-commit hook to enforce consistency across the SQL codebase. It will auto-format the files, a-la Prettier, when the hook runs

### Please do the following to test my code:

1. Stop and remove all docker containers, delete the `communitea-db` volume, and delete the `communitea-goapi` image
2. In the project's root directory, copy the contents of `.env.sample` into a `.env` file and replace any placeholder values (**NOTE:** Do **not** alter `.env.sample`)
3. Delete any `.env` files that are located outside of the root directory. Their contents should all be in the root directory's `.env` now
4. Apply changes to the docker config locally by navigating to the project directory in your terminal and running:
```
docker volume create communitea-db
docker compose build --no-cache
docker compose up
```
5. Ensure that all containers come up and stay live
6. Using your database management gui of choice (I recommend Beekeeper Studio), connect to the Postgres database and check that the location states and location cities tables have been migrated in. Additionally, confirm that all US states have auto-populated the location states table.


[Related Jira Ticket](https://communiteam.atlassian.net/browse/TEA-22)
